src/yuzu/main.cpp

```diff
@@ -1965,6 +1966,158 @@ void GMainWindow::OnGameListOpenFolder(u64 program_id, GameListOpenTarget target
+        ASSERT(user_id);
+
+        const auto user_save_data_path = FileSys::SaveDataFactory::GetFullPath(
+            *system, vfs_nand_dir, FileSys::SaveDataSpaceId::NandUser,
+            FileSys::SaveDataType::SaveData, program_id, user_id->AsU128(), 0);
+
+        path = Common::FS::ConcatPathSafe(nand_dir, user_save_data_path);
+    } else {
+        // Device save data
+        const auto device_save_data_path = FileSys::SaveDataFactory::GetFullPath(
+            *system, vfs_nand_dir, FileSys::SaveDataSpaceId::NandUser,
+            FileSys::SaveDataType::SaveData, program_id, {}, 0);
+
+        path = Common::FS::ConcatPathSafe(nand_dir, device_save_data_path);
+    }
+
+    const QString qpath = QString::fromStdString(Common::FS::PathToUTF8String(path));
+    const QDir dir(qpath);
+    QFile file;
+    QDataStream dataStream;
+
+    if (operation == 0) {
+        if (!dir.exists()) {
+            QMessageBox::warning(this, tr("Error Opening %1 Folder").arg(open_target),
+                                 tr("Folder does not exist!"));
+            return;
+        }
+
+        QString destinationFile = QFileDialog::getSaveFileName(
+            this, tr("Save Game Files"),
+            QString(QStandardPaths::DocumentsLocation) + QString::fromStdString("/"),
+            tr("Yuzu Data Save (*.yzsv)"));
+
+        file.setFileName(destinationFile);
+        file.open(QIODevice::WriteOnly);
+
+        dataStream.setDevice(&file);
+
+        QFileInfoList filesList = dir.entryInfoList();
+
+        for (int i = 0; i < filesList.length(); i++) {
+            qCritical() << "for loop run " << i;
+            QFile filefor(dir.absolutePath() + QString::fromStdString("/") +
+                          filesList.at(i).fileName());
+            qCritical() << "filefor " << filefor;
+            if (!filefor.open(QIODevice::ReadOnly))
+                continue;
+
+            dataStream << QString::fromStdString(fmt::format("{:016x}", program_id));
+            dataStream << filesList.at(i).fileName();
+            dataStream << filefor.readAll();
+
+            filefor.close();
+        }
+        file.close();
+    } else {
+        QString sourceFile = QFileDialog::getOpenFileName(
+            this, tr("Choose Game File"),
+            QString(QStandardPaths::DocumentsLocation) + QString::fromStdString("/"),
+            tr("Yuzu Data Save (*.yzsv)"));
+
+        QFile src(sourceFile);
+        if (!src.exists()) {
+            return;
+        }
+
+        file.setFileName(sourceFile);
+        if (!file.open(QIODevice::ReadOnly)) {
+            return;
+        }
+
+        dataStream.setDevice(&file);
+
+        while (!dataStream.atEnd()) {
+            QString fileName, game_id;
+            QByteArray dataByteArray;
+
+            dataStream >> game_id >> fileName >> dataByteArray;
+
+            QFile outFile(qpath + QString::fromStdString("/") + fileName);
+            if (!dir.exists()) {
```

Comment:
These if statements can be combined

```suggestion
            if (!dir.exists() && !dir.mkpath(qpath)) {
```